modules/ebs_snapshot_explorer/main.py

```diff
@@ -104,40 +128,68 @@ def get_valid_device(client, instance):
         str: Returns next mapping in form of /dev/xvd[base], otherwise /dev/xvdzz
 
     """
-    response = client.describe_instances(InstanceIds=[instance])
-    mappings = response['Reservations'][0]['Instances'][0]['BlockDeviceMappings']
+    instance = [instance for instance in get_instances(pacu) if instance['InstanceId'] == instance_id]
+    mappings = instance[0]['BlockDeviceMappings']
     current_mappings = [device['DeviceName'] for device in mappings]
-    base_mappings = [char for char in 'bcdefghijklmnoqrstuvwxyz']
-    for base in base_mappings:
-        if '/dev/xvd' + base not in current_mappings:
-            return '/dev/xvd' + base
-    return '/dev/xvdzz'
+    last_mapping = sorted(current_mappings)[-1]
+    available_devices = [get_valid_device(last_mapping)]
+    for _ in range(SET_COUNT):
+        available_devices.append(get_valid_device(available_devices[-1]))
+    return available_devices
+
+
+def get_valid_device(previous_device):
+    """Helper function that returns the next device given a previous device"""
+    return previous_device[:-1] + next_char(previous_device[-1])
 
 
-def get_snapshots(pacu, session, region):
+def next_char(char):
+    """Gets the next sequential character
+
+    Args:
+        char (str): Character to increment
+    Returns:
+        str: Incremented passed char
+    """
+    out = chr(ord(char) + 1)
+    return out if out != '{' else 'aa'
+
+
+def get_instances(pacu):
     """Returns snapshots given an AWS region
     Args:
         pacu (Main): Reference to Pacu
-        session (PacuSession): Reference to the Pacu session database
-        regions (list): Regions to check for snapshots
     Returns:
-        dict: Mapping regions to corresponding list of snapshot_ids.
+        list: List of Instances.
     """
-    ec2_data = deepcopy(session.EC2)
+    ec2_data = deepcopy(pacu.get_active_session().EC2)
+    if 'Instances' not in ec2_data:
+        fields = ['EC2', 'Instances']
+        module = 'enum_ec2'
+        fetched_ec2_instances = pacu.fetch_data(fields, module)
```

Comment:
Make sure to pass `--instances` to enum_ec2 so that you don't run the full enum_ec2 module